staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -577,7 +577,13 @@ func (s *SecureServingInfo) HostPort() (string, int, error) {
 // AuthorizeClientBearerToken wraps the authenticator and authorizer in loopback authentication logic
 // if the loopback client config is specified AND it has a bearer token.
 func AuthorizeClientBearerToken(loopback *restclient.Config, authn *AuthenticationInfo, authz *AuthorizationInfo) {
-	if loopback == nil || authn == nil || authz == nil || authn.Authenticator == nil && authz.Authorizer == nil || len(loopback.BearerToken) == 0 {
+	if loopback == nil || len(loopback.BearerToken) == 0 {
+		// do noting if either the loopback client config is nil or it doesn't have a bearer token.
+		return
+	}
+	if authn.Authenticator == nil || authz.Authorizer == nil {
```

Comment:
and if authn or authz is nil? We panic?

Reply:
it won't panic. authn/authz will just be bypassed

[https://github.com/kubernetes/kubernetes/blob/529785e35564d52b08b3ce303df5faf4055d6d31/staging/src/k8s.io/apiserver/pkg/endpoints/filters/authorization.go#L46-L49](https://github.com/kubernetes/kubernetes/blob/529785e35564d52b08b3ce303df5faf4055d6d31/staging/src/k8s.io/apiserver/pkg/endpoints/filters/authorization.go#L46-L49)

[https://github.com/kubernetes/kubernetes/blob/529785e35564d52b08b3ce303df5faf4055d6d31/staging/src/k8s.io/apiserver/pkg/endpoints/filters/authentication.go#L54-L57](https://github.com/kubernetes/kubernetes/blob/529785e35564d52b08b3ce303df5faf4055d6d31/staging/src/k8s.io/apiserver/pkg/endpoints/filters/authentication.go#L54-L57)

Reply:
But here we check the `Authn/zInfo` structs, not the actual authenticators/authorizers. These info pointers can be nil. Then this code here panics.

Reply:
👍🏻updated